### PR TITLE
add Lua info

### PR DESCRIPTION
**Pull request by borbware**
(second try)

### CODE REVIEW

File: which_language.md [hunk 1]
```diff
@@ -190,3 +190,46 @@ Materials
 - [You Don't Know JS](https://github.com/getify/You-Dont-Know-JS)
 - [JS for C & Python programmers](https://www.wooji-juice.com/blog/javascript-article.html)
 
+Lua
+==========
+A light, somewhat verbose Python-like language where indentation doesn't matter and almost everything is a table. Also, indexing starts from one. Typically, the Lua VM sits on top of a lower-level engine, providing an easy-to-use scripting interface.
```
Comment (ronchaine) [inline which_language.md]: Could you please break the lines so their length is around 80 columns max?

Comment (ronchaine) [inline which_language.md]: Python-like could be misleading here.  I see what you mean but if the target audience is people who do not know what "a table" is, this is overly detailed.  Same with "indexing starts from one", as I would hope reading the intro doesn't require too much (any) technical knowledge.

Also it shouldn't be required to know what a "VM" is to read the intro

File: which_language.md [hunk 2]
```diff
@@ -190,3 +190,46 @@ Materials
...
+-----------
+- video game scripting & modding
+- game development (Love2D, Pico-8, TIC-80)
+  
+Pros
+----
+
+- tiny & portable
+- short learning curve
+- fast, especially when using the JIT compiler
```
Comment (ronchaine) [inline which_language.md]: A bit questionable/misleading to call Lua fast, when there are things like C, C++, Java and C# in the same list.

Also, again.  Do not assume reader knows what a "JIT compiler" is.

File: which_language.md [hunk 3]
```diff
@@ -190,3 +190,46 @@ Materials
...
+- powerful table system supports multiple key & value types
+  - ...and can be further extended with metatables
```
Comment (ronchaine) [inline which_language.md]: Same here, expects too much from the reader.  "Table system", "key & value types" and "metatables".

File: which_language.md [hunk 4]
```diff
@@ -190,3 +190,46 @@ Materials
...
+- functions:
+  - inline functions don't have a confusing lambda syntax (looking at you, Python)
```
Comment (ronchaine) [inline which_language.md]: Assume reader doesn't know what "Lambda" is

Also, please no bashing of other languages in the list (even if it was kinda true)

File: which_language.md [hunk 5]
```diff
@@ -190,3 +190,46 @@ Materials
...
+  - can be tail called, recursed & passed around easily
```
Comment (ronchaine) [inline which_language.md]: Again, none of this makes sense to somebody with no programming experience.

File: which_language.md [hunk 6]
```diff
@@ -190,3 +190,46 @@ Materials
...
+  - multi-value return & variable number of function parameters
+- coroutines!
+
+Cons
+----
+- the Lua package ecosystem is a mess
```
Comment (ronchaine) [inline which_language.md]: Don't assume reader knows what Ecosystem is

File: which_language.md [hunk 7]
```diff
@@ -190,3 +190,46 @@ Materials
...
+- the community is split between different Lua versions
+- pattern-matching that is almost regex, but not quite
```
Comment (ronchaine) [inline which_language.md]: Don't assume knowledge of "pattern matching" and "regex"

File: which_language.md [hunk 8]
```diff
@@ -190,3 +190,46 @@ Materials
...
+- tiny size has downsides (depends on who you ask):
+  - for a language obsessed with tables, you'd think there were more table-specific features built in (like printing a table)
+  - no object oriented features out of the box
+  - limited unicode support
```
Comment (ronchaine) [inline which_language.md]: Not sure if "Unicode" is in the list of "don't assume knowledge", your call.

**Review by ronchaine — CHANGES_REQUESTED** [review]
I think this expects too much from the reader.  Assume zero prior knowledge.

Explanation should be somewhat higher-level.

File: which_language.md [hunk 9]
```diff
@@ -190,3 +190,38 @@ Materials
...
+easy-to-use scripting interface.
+
+Usual usage
+-----------
+- video game scripting & modding
+- game development (Love2D, Pico-8, TIC-80)
+  
+Pros
+----
+
```
Comment (ronchaine) [inline which_language.md:206]: Extra line.

**Review by ronchaine — APPROVED** [review]
LGTM.  I can fix the formatting or you can do it yourself.  However you like.

File: which_language.md (hunk 9 above)
Comment (borbware) [reply which_language.md:206]: fix away

**Pull request merged**